lib/ecto/adapters/sql.ex

```diff
@@ -355,6 +355,28 @@ defmodule Ecto.Adapters.SQL do
     opts
   end
 
+  @doc """
+  Check if the given `table` exists in the given `database`.
+
+  Returns `true` if the `table` exists in the `database` using the given
+  `adapter_meta`, otherwise `false`.
+  """
+  @spec table_exists?(Ecto.Repo.t | Ecto.Adapter.adapter_meta, table :: String.t, schema :: String.t) :: boolean
+  def table_exists?(repo_or_adapter, table, schema) do
```

Comment:
MySQL is only failing because `public` is not the default schema.

Any ideas of how I can discover the default configured schema?

Reply:
In MySQL the schema is the database name. However, we don't call it schema in Ecto, we call it "prefix". My suggestion would be to find a way to run this query without requiring a SCHEMA (or ask the database to return the current schema) in the query itself.